components/autocomplete/index.js

```diff
@@ -169,48 +329,87 @@ class Autocomplete extends Component {
 				this.select( options[ selectedIndex ] );
 				break;
 
+			case LEFT:
+			case RIGHT:
+			case TAB:
+				this.reset();
+				return;
+
 			default:
 				return;
 		}
 
 		// Any handled keycode should prevent original behavior. This relies on
 		// the early return in the default case.
 		event.preventDefault();
-		event.stopImmediatePropagation();
+		event.stopPropagation();
+	}
+
+	toggleKeyEvents( isListening ) {
+		// This exists because we must capture ENTER key presses before Editable.
+		// It seems that react fires the simulated capturing events after the
+		// native browser event has already bubbled so we can't stopPropagation
+		//  and avoid Editable getting the event from TinyMCE, hence we must
+		// register a native event handler.
+		if ( ! this.node ) {
+			return;
+		}
+		// Disable reason: We need to add native event handlers because they must
+		// be able to cancel the event before TinyMCE's native event handler gets it.
+		// eslint-disable-next-line react/no-find-dom-node
+		const realNode = findDOMNode( this.node );
+		const handler = isListening ? 'addEventListener' : 'removeEventListener';
+		realNode[ handler ]( 'keydown', this.setSelectedIndex, true );
+	}
+
+	componentDidUpdate( prevProps, prevState ) {
+		const { open } = this.state;
+		const { open: prevOpen } = prevState;
+		if ( ( ! open ) !== ( ! prevOpen ) ) {
+			this.toggleKeyEvents( ! ! open );
+		}
+	}
+
+	componentWillUnmount() {
+		this.toggleKeyEvents( false );
 	}
 
 	render() {
-		const { children, className } = this.props;
-		const { isOpen, selectedIndex } = this.state;
+		const { children, instanceId } = this.props;
+		const { open, selectedIndex, range } = this.state;
+		const { className } = open || {};
 		const classes = classnames( 'components-autocomplete__popover', className );
 		const filteredOptions = this.getFilteredOptions();
+		const isOpen = filteredOptions.length > 0;
+		const listBoxId = isOpen ? `components-autocomplete-listbox-${ instanceId }` : null;
+		const activeId = isOpen ? `components-autocomplete-item-${ instanceId }-${ selectedIndex }` : null;
 
-		// Blur is applied to the wrapper node, since if the child is Editable,
-		// the event will not have `relatedTarget` assigned.
 		return (
 			<div
-				onBlur={ this.onBlur }
+				ref={ this.bindNode }
+				onInput={ this.search }
 				className="components-autocomplete"
 			>
-				{ cloneElement( Children.only( children ), {
-					onInput: this.search,
```

Comment:
Since we added Editable "event proxying" specifically to allow for this, and given some of the challenges we've encountered with the event proxying, I'm wondering if now that we're no longer overriding these props that we drop the Editable logic:

https://github.com/WordPress/gutenberg/blob/ce7870c821a4e80c2df27da2aa17af64dfc7eae4/blocks/editable/index.js#L146-L163

Reply:
Fine with me but should probably be in another PR.